I dont really know what is going on here but it crashes every time i put it on a server

this is the error i get

[17:38:58] [main/ERROR]: Failed to start the minecraft server
--
78 | java.lang.RuntimeException: Could not execute entrypoint stage 'main' due to errors, provided by 'forgero'!
79 | at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.lambda$invoke0$0(EntrypointUtils.java:51) ~[fabric-loader-0.14.9.jar:?]
80 | at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33) ~[fabric-loader-0.14.9.jar:?]
81 | at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:49) ~[fabric-loader-0.14.9.jar:?]
82 | at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke(EntrypointUtils.java:35) ~[fabric-loader-0.14.9.jar:?]
83 | at net.fabricmc.loader.impl.game.minecraft.Hooks.startServer(Hooks.java:62) ~[fabric-loader-0.14.9.jar:?]
84 | at net.minecraft.server.Main.main(Main.java:101) [server-intermediary.jar:?]
85 | at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) [fabric-loader-0.14.9.jar:?]
86 | at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.14.9.jar:?]
87 | at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23) [fabric-loader-0.14.9.jar:?]
88 | at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69) [fabric-loader-0.14.9.jar:?]
89 | Caused by: java.util.NoSuchElementException: No value present
90 | at java.util.Optional.get(Optional.java:143) ~[?:?]
91 | at net.minecraft.class_1766.handler$zkk000$init(class_1766.java:1543) ~[server-intermediary.jar:?]
92 | at net.minecraft.class_1766.<init>(class_1766.java:34) ~[server-intermediary.jar:?]
93 | at net.minecraft.class_1821.<init>(class_1821.java:35) ~[server-intermediary.jar:?]
94 | at com.sigmundgranaas.forgero.item.items.tool.ForgeroShovelItem.<init>(ForgeroShovelItem.java:34) ~[forgero-0.8.4-BETA-1.19.0.jar:?]
95 | at com.sigmundgranaas.forgero.item.implementation.ItemFactoryImpl.createTool(ItemFactoryImpl.java:45) ~[forgero-0.8.4-BETA-1.19.0.jar:?]
96 | at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?]
97 | at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:992) ~[?:?]
98 | at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
99 | at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
100 | at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575) ~[?:?]
101 | at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260) ~[?:?]
102 | at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616) ~[?:?]
103 | at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622) ~[?:?]
104 | at java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627) ~[?:?]
105 | at com.sigmundgranaas.forgero.registry.impl.ConcurrentForgeroItemRegistry.loadResources(ConcurrentForgeroItemRegistry.java:55) ~[forgero-0.8.4-BETA-1.19.0.jar:?]
106 | at com.sigmundgranaas.forgero.registry.impl.ConcurrentForgeroItemRegistry.loadResourcesIfEmpty(ConcurrentForgeroItemRegistry.java:68) ~[forgero-0.8.4-BETA-1.19.0.jar:?]
107 | at com.sigmundgranaas.forgero.ForgeroInitializer.onInitialize(ForgeroInitializer.java:30) ~[forgero-0.8.4-BETA-1.19.0.jar:?]
108 | at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:47) ~[fabric-loader-0.14.9.jar:?]
109 | ... 7 more

<!--EndFragment-->
</body>
</html>[17:38:58] [main/ERROR]: Failed to start the minecraft server
java.lang.RuntimeException: Could not execute entrypoint stage 'main' due to errors, provided by 'forgero'!
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.lambda$invoke0$0(EntrypointUtils.java:51) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:49) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke(EntrypointUtils.java:35) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startServer(Hooks.java:62) ~[fabric-loader-0.14.9.jar:?]
	at net.minecraft.server.Main.main(Main.java:101) [server-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) [fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23) [fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69) [fabric-loader-0.14.9.jar:?]
Caused by: java.util.NoSuchElementException: No value present
	at java.util.Optional.get(Optional.java:143) ~[?:?]
	at net.minecraft.class_1766.handler$zkk000$init(class_1766.java:1543) ~[server-intermediary.jar:?]
	at net.minecraft.class_1766.<init>(class_1766.java:34) ~[server-intermediary.jar:?]
	at net.minecraft.class_1[82](https://aternos.org/log/#L82)1.<init>(class_1821.java:35) ~[server-intermediary.jar:?]
	at com.sigmundgranaas.forgero.item.items.tool.ForgeroShovelItem.<init>(ForgeroShovelItem.java:34) ~[forgero-0.8.4-BETA-1.19.0.jar:?]
	at com.sigmundgranaas.forgero.item.implementation.ItemFactoryImpl.createTool(ItemFactoryImpl.java:45) ~[forgero-0.8.4-BETA-1.19.0.jar:?]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?]
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:9[92](https://aternos.org/log/#L92)) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:4[99](https://aternos.org/log/#L99)) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260) ~[?:?]
	at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616) ~[?:?]
	at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622) ~[?:?]
	at java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627) ~[?:?]
	at com.sigmundgranaas.forgero.registry.impl.ConcurrentForgeroItemRegistry.loadResources(ConcurrentForgeroItemRegistry.java:55) ~[forgero-0.8.4-BETA-1.19.0.jar:?]
	at com.sigmundgranaas.forgero.registry.impl.ConcurrentForgeroItemRegistry.loadResourcesIfEmpty(ConcurrentForgeroItemRegistry.java:68) ~[forgero-0.8.4-BETA-1.19.0.jar:?]
	at com.sigmundgranaas.forgero.ForgeroInitializer.onInitialize(ForgeroInitializer.java:30) ~[forgero-0.8.4-BETA-1.19.0.jar:?]
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:47) ~[fabric-loader-0.14.9.jar:?]
	... 7 more

does anyone know why?

Loading 52 mods:
9 | - advanced_runtime_resource_pack 0.6.2 via forgero
10 | - allstackable v1.5.6-1.19
11 | - camp_fires_cook_mobs 1.2.9
12 | - cloth-basic-math 0.6.1 via cloth-config
13 | - cloth-config 8.0.75
14 | - clumps 9.0.0+7
15 | - collective-fabric 4.36
16 | - combatedit 1.6.1
17 | - crowdin-translate 1.4+1.19 via camp_fires_cook_mobs
18 | - fabric-api 0.59.0+1.19.2
19 | - fabric-api-base 0.4.10+e62f51a390 via fabric-api
20 | - fabric-api-lookup-api-v1 1.6.8+9ff28f4090 via fabric-api
21 | - fabric-biome-api-v1 9.0.16+83a8659290 via fabric-api
22 | - fabric-command-api-v1 1.2.10+f71b366f90 via fabric-api
23 | - fabric-command-api-v2 2.1.6+0c17ea9690 via fabric-api
24 | - fabric-commands-v0 0.2.27+df3654b390 via fabric-api
25 | - fabric-containers-v0 0.1.33+df3654b390 via fabric-api
26 | - fabric-content-registries-v0 3.2.2+83a8659290 via fabric-api
27 | - fabric-convention-tags-v1 1.1.0+7cd20a1490 via fabric-api
28 | - fabric-crash-report-info-v1 0.2.4+bd0a0d4a90 via fabric-api
29 | - fabric-data-generation-api-v1 5.1.7+a680b9b490 via fabric-api
30 | - fabric-dimensions-v1 2.1.29+a6d2f78590 via fabric-api
31 | - fabric-entity-events-v1 1.4.17+9ff28f4090 via fabric-api
32 | - fabric-events-interaction-v0 0.4.27+9ff28f4090 via fabric-api
33 | - fabric-events-lifecycle-v0 0.2.27+df3654b390 via fabric-api
34 | - fabric-game-rule-api-v1 1.0.20+1899036190 via fabric-api
35 | - fabric-item-api-v1 1.5.6+35a03c4390 via fabric-api
36 | - fabric-item-groups-v0 0.3.27+6bee109e90 via fabric-api
37 | - fabric-lifecycle-events-v1 2.1.1+83a8659290 via fabric-api
38 | - fabric-loot-api-v2 1.1.2+83a8659290 via fabric-api
39 | - fabric-loot-tables-v1 1.1.5+9e7660c690 via fabric-api
40 | - fabric-message-api-v1 5.0.2+176380a290 via fabric-api
41 | - fabric-mining-level-api-v1 2.1.13+33fbc73890 via fabric-api
42 | - fabric-networking-api-v1 1.2.3+5eb68ef290 via fabric-api
43 | - fabric-networking-v0 0.3.20+df3654b390 via fabric-api
44 | - fabric-object-builder-api-v1 4.0.10+7675279690 via fabric-api
45 | - fabric-particles-v1 1.0.9+dc39553a90 via fabric-api
46 | - fabric-registry-sync-v0 0.9.23+23c4cfef90 via fabric-api
47 | - fabric-rendering-data-attachment-v1 0.3.13+9ff28f4090 via fabric-api
48 | - fabric-rendering-fluids-v1 3.0.6+9ff28f4090 via fabric-api
49 | - fabric-resource-conditions-api-v1 2.0.10+e62f51a390 via fabric-api
50 | - fabric-resource-loader-v0 0.6.0+6bee109e90 via fabric-api
51 | - fabric-screen-handler-api-v1 1.2.12+9ff28f4090 via fabric-api
52 | - fabric-transfer-api-v1 2.0.10+83a8659290 via fabric-api
53 | - fabric-transitive-access-wideners-v1 1.2.0+1e232e1390 via fabric-api
54 | - fabricloader 0.14.9
55 | - fantasy 0.4.6+1.19.2 via multiworld
56 | - fiber 0.23.0-2 via collective-fabric
57 | - forgero 0.9.5-BETA-1.19.2
58 | - java 17
59 | - minecraft 1.19.1
60 | - multiworld 1.5
61 | [12:40:50] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.5 Source=file:/server/libraries/net/fabricmc/sponge-mixin/0.11.4+mixin.0.8.5/sponge-mixin-0.11.4+mixin.0.8.5.jar Service=Knot/Fabric Env=SERVER
62 | [12:40:50] [main/INFO]: Compatibility level set to JAVA_17
63 | [12:40:52] [main/INFO]: I used the json to destroy the json
64 | [12:40:55] [main/INFO]: forgero.mixins.json:ItemUUIDMixin from mod forgero->@Accessor[FIELD_GETTER]::getATTACK_DAMAGE_MODIFIER_ID()Ljava/util/UUID; should be static as its target is
65 | [12:41:02] [main/INFO]: Building unoptimized datafixer
66 | [12:41:06] [main/INFO]: [All Stackable] Start loading!
67 | [12:41:06] [main/INFO]: [All Stackable] Command registered.
68 | [12:41:06] [main/INFO]: [STDOUT]: Loading Collective (Fabric) version 4.36.
69 | [12:41:06] [main/INFO]: [CombatEdit] Successfully initialized.
70 | [12:41:07] [main/INFO]: [STDOUT]:  Multiworld init
71 | [12:41:07] [main/ERROR]: Failed to start the minecraft server
72 | java.lang.RuntimeException: Could not execute entrypoint stage 'main' due to errors, provided by 'forgero'!
73 | at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.lambda$invoke0$0(EntrypointUtils.java:51) ~[fabric-loader-0.14.9.jar:?]
74 | at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33) ~[fabric-loader-0.14.9.jar:?]
75 | at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:49) ~[fabric-loader-0.14.9.jar:?]
76 | at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke(EntrypointUtils.java:35) ~[fabric-loader-0.14.9.jar:?]
77 | at net.fabricmc.loader.impl.game.minecraft.Hooks.startServer(Hooks.java:62) ~[fabric-loader-0.14.9.jar:?]
78 | at net.minecraft.server.Main.main(Main.java:101) [server-intermediary.jar:?]
79 | at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) [fabric-loader-0.14.9.jar:?]
80 | at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.14.9.jar:?]
81 | at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23) [fabric-loader-0.14.9.jar:?]
82 | at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69) [fabric-loader-0.14.9.jar:?]
83 | Caused by: java.util.NoSuchElementException: No value present
84 | at java.util.Optional.get(Optional.java:143) ~[?:?]
85 | at net.minecraft.class_1766.handler$zkd000$init(class_1766.java:1543) ~[server-intermediary.jar:?]
86 | at net.minecraft.class_1766.<init>(class_1766.java:34) ~[server-intermediary.jar:?]
87 | at net.minecraft.class_1743.<init>(class_1743.java:50) ~[server-intermediary.jar:?]
88 | at com.sigmundgranaas.forgero.item.items.tool.ForgeroAxeItem.<init>(ForgeroAxeItem.java:35) ~[forgero-0.9.5-BETA-1.19.2.jar:?]
89 | at com.sigmundgranaas.forgero.item.implementation.ItemFactoryImpl.createTool(ItemFactoryImpl.java:46) ~[forgero-0.9.5-BETA-1.19.2.jar:?]
90 | at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?]
91 | at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:992) ~[?:?]
92 | at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
93 | at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
94 | at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575) ~[?:?]
95 | at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260) ~[?:?]
96 | at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616) ~[?:?]
97 | at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622) ~[?:?]
98 | at java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627) ~[?:?]
99 | at com.sigmundgranaas.forgero.registry.impl.ConcurrentForgeroItemRegistry.loadResources(ConcurrentForgeroItemRegistry.java:55) ~[forgero-0.9.5-BETA-1.19.2.jar:?]
100 | at com.sigmundgranaas.forgero.registry.impl.ConcurrentForgeroItemRegistry.loadResourcesIfEmpty(ConcurrentForgeroItemRegistry.java:68) ~[forgero-0.9.5-BETA-1.19.2.jar:?]
101 | at com.sigmundgranaas.forgero.ForgeroInitializer.onInitialize(ForgeroInitializer.java:30) ~[forgero-0.9.5-BETA-1.19.2.jar:?]
102 | at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:47) ~[fabric-loader-0.14.9.jar:?]
103 | ... 7 more
104

<br class="Apple-interchange-newline"><!--EndFragment-->
</body>
</html>Loading 52 mods:
	- advanced_runtime_resource_pack 0.6.2 via forgero
	- allstackable v1.5.6-1.1[9](https://aternos.org/log/#L9)
	- camp_fires_cook_mobs 1.2.9
	- cloth-basic-math 0.6.1 via cloth-config
	- cloth-config 8.0.75
	- clumps 9.0.0+7
	- collective-fabric 4.36
	- combatedit 1.6.1
	- crowdin-translate 1.4+1.19 via camp_fires_cook_mobs
	- fabric-api 0.59.0+1.19.2
	- fabric-api-base 0.4.[10](https://aternos.org/log/#L10)+e62f51a390 via fabric-api
	- fabric-api-lookup-api-v1 1.6.8+9ff28f4090 via fabric-api
	- fabric-biome-api-v1 9.0.16+83a8659290 via fabric-api
	- fabric-command-api-v1 1.2.10+f71b366f90 via fabric-api
	- fabric-command-api-v2 2.1.6+0c17ea9690 via fabric-api
	- fabric-commands-v0 0.2.27+df3654b390 via fabric-api
	- fabric-containers-v0 0.1.33+df3654b390 via fabric-api
	- fabric-content-registries-v0 3.2.2+83a8659290 via fabric-api
	- fabric-convention-tags-v1 1.1.0+7cd20a1490 via fabric-api
	- fabric-crash-report-info-v1 0.2.4+bd0a0d4a90 via fabric-api
	- fabric-data-generation-api-v1 5.1.7+a680b9b490 via fabric-api
	- fabric-dimensions-v1 2.1.29+a6d2f78590 via fabric-api
	- fabric-entity-events-v1 1.4.17+9ff28f4090 via fabric-api
	- fabric-events-interaction-v0 0.4.27+9ff28f4090 via fabric-api
	- fabric-events-lifecycle-v0 0.2.27+df3654b390 via fabric-api
	- fabric-game-rule-api-v1 1.0.20+1899036190 via fabric-api
	- fabric-item-api-v1 1.5.6+35a03c4390 via fabric-api
	- fabric-item-groups-v0 0.3.27+6bee109e90 via fabric-api
	- fabric-lifecycle-events-v1 2.1.1+83a8659290 via fabric-api
	- fabric-loot-api-v2 1.1.2+83a8659290 via fabric-api
	- fabric-loot-tables-v1 1.1.5+9e7660c690 via fabric-api
	- fabric-message-api-v1 5.0.2+176380a290 via fabric-api
	- fabric-mining-level-api-v1 2.1.13+33fbc73890 via fabric-api
	- fabric-networking-api-v1 1.2.3+5eb68ef290 via fabric-api
	- fabric-networking-v0 0.3.20+df3654b390 via fabric-api
	- fabric-object-builder-api-v1 4.0.10+7675279690 via fabric-api
	- fabric-particles-v1 1.0.9+dc39553a90 via fabric-api
	- fabric-registry-sync-v0 0.9.23+23c4cfef90 via fabric-api
	- fabric-rendering-data-attachment-v1 0.3.13+9ff28f4090 via fabric-api
	- fabric-rendering-fluids-v1 3.0.6+9ff28f4090 via fabric-api
	- fabric-resource-conditions-api-v1 2.0.10+e62f51a390 via fabric-api
	- fabric-resource-loader-v0 0.6.0+6bee109e90 via fabric-api
	- fabric-screen-handler-api-v1 1.2.12+9ff28f4090 via fabric-api
	- fabric-transfer-api-v1 2.0.10+83a8659290 via fabric-api
	- fabric-transitive-access-wideners-v1 1.2.0+1e232e1390 via fabric-api
	- fabricloader 0.14.9
	- fantasy 0.4.6+1.19.2 via multiworld
	- fiber 0.23.0-2 via collective-fabric
	- forgero 0.9.5-BETA-1.19.2
	- java 17
	- minecraft 1.19.1
	- multiworld 1.5
[12:40:50] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.5 Source=file:/server/libraries/net/fabricmc/sponge-mixin/0.[11](https://aternos.org/log/#L11).4+mixin.0.8.5/sponge-mixin-0.11.4+mixin.0.8.5.jar Service=Knot/Fabric Env=SERVER
[[12](https://aternos.org/log/#L12):40:50] [main/INFO]: Compatibility level set to JAVA_17
[12:40:52] [main/INFO]: I used the json to destroy the json
[12:40:55] [main/INFO]: forgero.mixins.json:ItemUUIDMixin from mod forgero->@Accessor[FIELD_GETTER]::getATTACK_DAMAGE_MODIFIER_ID()Ljava/util/UUID; should be static as its target is
[12:41:02] [main/INFO]: Building unoptimized datafixer
[12:41:06] [main/INFO]: [All Stackable] Start loading!
[12:41:06] [main/INFO]: [All Stackable] Command registered.
[12:41:06] [main/INFO]: [STDOUT]: Loading Collective (Fabric) version 4.36.
[12:41:06] [main/INFO]: [CombatEdit] Successfully initialized.
[12:41:07] [main/INFO]: [STDOUT]:  Multiworld init
[12:41:07] [main/ERROR]: Failed to start the minecraft server
java.lang.RuntimeException: Could not execute entrypoint stage 'main' due to errors, provided by 'forgero'!
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.lambda$invoke0$0(EntrypointUtils.java:51) ~[fabric-loader-0.[14](https://aternos.org/log/#L14).9.jar:?]
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:49) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke(EntrypointUtils.java:35) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startServer(Hooks.java:62) ~[fabric-loader-0.14.9.jar:?]
	at net.minecraft.server.Main.main(Main.java:101) [server-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) [fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23) [fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69) [fabric-loader-0.14.9.jar:?]
Caused by: java.util.NoSuchElementException: No value present
	at java.util.Optional.get(Optional.java:143) ~[?:?]
	at net.minecraft.class_1766.handler$zkd000$init(class_1766.java:[15](https://aternos.org/log/#L15)43) ~[server-intermediary.jar:?]
	at net.minecraft.class_1766.<init>(class_1766.java:34) ~[server-intermediary.jar:?]
	at net.minecraft.class_1743.<init>(class_1743.java:50) ~[server-intermediary.jar:?]
	at com.sigmundgranaas.forgero.item.items.tool.ForgeroAxeItem.<init>(ForgeroAxeItem.java:35) ~[forgero-0.9.5-BETA-1.19.2.jar:?]
	at com.sigmundgranaas.forgero.item.implementation.ItemFactoryImpl.createTool(ItemFactoryImpl.java:46) ~[forgero-0.9.5-BETA-1.19.2.jar:?]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?]
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:992) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260) ~[?:?]
	at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:6[16](https://aternos.org/log/#L16)) ~[?:?]
	at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622) ~[?:?]
	at java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627) ~[?:?]
	at com.sigmundgranaas.forgero.registry.impl.ConcurrentForgeroItemRegistry.loadResources(ConcurrentForgeroItemRegistry.java:55) ~[forgero-0.9.5-BETA-1.[19](https://aternos.org/log/#L19).2.jar:?]
	at com.sigmundgranaas.forgero.registry.impl.ConcurrentForgeroItemRegistry.loadResourcesIfEmpty(ConcurrentForgeroItemRegistry.java:68) ~[forgero-0.9.5-BETA-1.19.2.jar:?]
	at com.sigmundgranaas.forgero.ForgeroInitializer.onInitialize(ForgeroInitializer.java:[30](https://aternos.org/log/#L30)) ~[forgero-0.9.5-BETA-1.19.2.jar:?]
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:[47](https://aternos.org/log/#L47)) ~[fabric-loader-0.14.9.jar:?]
	... 7 more





it did still crash though and it said the error is  from line 71 onward.

--- a/build.gradle
+++ b/build.gradle
@@ -83,6 +83,8 @@ dependencies {
     //modImplementation "curse.maven:cardinal-components-318449:3669773"
     //modImplementation "curse.maven:polymorph-397434:3793108"
 
+    modImplementation "curse.maven:combatedit-385513:3869344"
+
     //modImplementation 'com.ptsmods:devlogin:2.1'
     // Fabric API. This is technically optional, but you probably want it anyway.
     modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"

--- a/changelog.md
+++ b/changelog.md
@@ -1,6 +1,14 @@
-### Changelog 0.9.5
+### Changelog 0.9.6
 
-# Forgero Beta release
+# Forgero Beta release 0.9.x
+
+-- 0.9.6 --
+
+* Fixed compatability with some resource packs
+* Updated ender gem and diamond gem to fix stats
+* Fixed crash when used with combatedit
+
+-- 0.9 --
 
 * Fixed damage upgrades affecting tool negatively instad of boosting stats.
 * Fixed Gem levels not being applied in ToolPart state

--- a/gradle.properties
+++ b/gradle.properties
@@ -5,7 +5,7 @@ minecraft_version=1.19.2
 yarn_mappings=1.19.2+build.1
 loader_version=0.14.8
 # Mod Properties
-mod_version=0.9.5-BETA-1.19.2
+mod_version=0.9.6-BETA-1.19.2
 maven_group=com.sigmundgranaas
 archives_base_name=forgero
 # Dependencies

--- a/src/main/java/com/sigmundgranaas/forgero/client/ForgeroClient.java
+++ b/src/main/java/com/sigmundgranaas/forgero/client/ForgeroClient.java
@@ -28,7 +28,8 @@ public class ForgeroClient implements ClientModInitializer {
     private void initializeItemModels() {
         new ForgeroTextures().createTextureIdentifiers();
         registerToolPartTextures();
-        ModelLoadingRegistry.INSTANCE.registerVariantProvider(variant -> new ForgeroModelVariantProvider(UnbakedModelCollection.INSTANCE));
+        var modelProvider = new ForgeroModelVariantProvider(UnbakedModelCollection.INSTANCE);
+        ModelLoadingRegistry.INSTANCE.registerVariantProvider(variant -> modelProvider);
     }
 
     private void registerToolPartTextures() {

--- a/src/main/java/com/sigmundgranaas/forgero/client/forgerotool/model/ForgeroModelVariantProvider.java
+++ b/src/main/java/com/sigmundgranaas/forgero/client/forgerotool/model/ForgeroModelVariantProvider.java
@@ -24,7 +24,6 @@ public class ForgeroModelVariantProvider implements ModelVariantProvider {
     UnbakedModel loadModelVariant(ModelIdentifier modelId, ModelProviderContext context) {
         if (modelId.getNamespace().equals(ForgeroInitializer.MOD_NAMESPACE) && !modelId.getPath().contains("transparent_base")) {
             String[] elements = modelId.getPath().split(DEFAULT_SPLIT_OPERATOR);
-
             if (elements.length > 1 && ForgeroRegistry.TOOL.resourceExists(modelId.getPath())) {
                 return toolModelVariant;
             } else if (elements.length == 2 && ForgeroRegistry.TOOL_PART.resourceExists(modelId.getPath())) {
@@ -34,4 +33,5 @@ public class ForgeroModelVariantProvider implements ModelVariantProvider {
         }
         return null;
     }
+
 }

--- a/src/main/java/com/sigmundgranaas/forgero/client/forgerotool/model/ToolModelVariant.java
+++ b/src/main/java/com/sigmundgranaas/forgero/client/forgerotool/model/ToolModelVariant.java
@@ -4,6 +4,7 @@ import com.google.common.cache.CacheBuilder;
 import com.google.common.cache.CacheLoader;
 import com.google.common.cache.LoadingCache;
 import com.sigmundgranaas.forgero.client.forgerotool.model.implementation.EmptyBakedModelCollection;
+import com.sigmundgranaas.forgero.item.ForgeroToolItem;
 import net.fabricmc.fabric.api.renderer.v1.model.FabricBakedModel;
 import net.fabricmc.fabric.api.renderer.v1.render.RenderContext;
 import net.minecraft.client.render.model.BakedModel;
@@ -11,6 +12,7 @@ import net.minecraft.client.render.model.ModelBakeSettings;
 import net.minecraft.client.render.model.ModelLoader;
 import net.minecraft.client.texture.Sprite;
 import net.minecraft.client.util.SpriteIdentifier;
+import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.Identifier;
 import net.minecraft.util.math.random.Random;
@@ -22,8 +24,10 @@ import java.util.function.Supplier;
 
 public class ToolModelVariant extends ForgeroCustomModelProvider {
     private final UnbakedModelCollection unbakedModelCollection;
-    private BakedModelCollection bakedModelCollection;
     private final LoadingCache<ItemStack, FabricBakedModel> cache;
+    private final LoadingCache<Item, FabricBakedModel> itemCache;
+    private BakedModelCollection bakedModelCollection;
+    private ModelLoader loader;
 
     public ToolModelVariant(UnbakedModelCollection collection) {
         this.unbakedModelCollection = collection;
@@ -34,18 +38,33 @@ public class ToolModelVariant extends ForgeroCustomModelProvider {
                 return bakedModelCollection.getToolModel(itemStack);
             }
         });
+        this.itemCache = CacheBuilder.newBuilder().maximumSize(600).build(new CacheLoader<>() {
+            @Override
+            public @NotNull FabricBakedModel load(@NotNull Item item) {
+                return bakedModelCollection.getToolModel(((ForgeroToolItem) item).getTool());
+            }
+        });
     }
 
     @Override
     public void emitItemQuads(ItemStack stack, Supplier<Random> randomSupplier, RenderContext context) {
+        if (stack.hasNbt()) {
             cache.getUnchecked(stack).emitItemQuads(null, null, context);
+        } else {
+            itemCache.getUnchecked(stack.getItem()).emitItemQuads(null, null, context);
+        }
+
     }
 
     @Nullable
     @Override
     public BakedModel bake(ModelLoader loader, Function<SpriteIdentifier, Sprite> textureGetter, ModelBakeSettings rotationContainer, Identifier modelId) {
         this.bakedModelCollection = unbakedModelCollection.bakeModels(loader, textureGetter);
+        if (this.loader == null || this.loader != loader) {
+            this.loader = loader;
+            cache.invalidateAll();
+            itemCache.invalidateAll();
+        }
         return this;
     }
-
 }

--- a/src/main/java/com/sigmundgranaas/forgero/client/forgerotool/model/ToolPartModelVariant.java
+++ b/src/main/java/com/sigmundgranaas/forgero/client/forgerotool/model/ToolPartModelVariant.java
@@ -4,6 +4,7 @@ import com.google.common.cache.CacheBuilder;
 import com.google.common.cache.CacheLoader;
 import com.google.common.cache.LoadingCache;
 import com.sigmundgranaas.forgero.client.forgerotool.model.implementation.EmptyBakedModelCollection;
+import com.sigmundgranaas.forgero.item.ToolPartItem;
 import net.fabricmc.fabric.api.renderer.v1.model.FabricBakedModel;
 import net.fabricmc.fabric.api.renderer.v1.render.RenderContext;
 import net.minecraft.client.render.model.BakedModel;
@@ -11,6 +12,7 @@ import net.minecraft.client.render.model.ModelBakeSettings;
 import net.minecraft.client.render.model.ModelLoader;
 import net.minecraft.client.texture.Sprite;
 import net.minecraft.client.util.SpriteIdentifier;
+import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.Identifier;
 import net.minecraft.util.math.random.Random;
@@ -22,29 +24,47 @@ import java.util.function.Supplier;
 
 public class ToolPartModelVariant extends ForgeroCustomModelProvider {
     private final UnbakedModelCollection unbakedModelCollection;
-    private BakedModelCollection bakedModelCollection;
     private final LoadingCache<ItemStack, FabricBakedModel> cache;
+    private final LoadingCache<Item, FabricBakedModel> itemCache;
+    private BakedModelCollection bakedModelCollection;
+    private ModelLoader loader;
 
     public ToolPartModelVariant(UnbakedModelCollection collection) {
         this.unbakedModelCollection = collection;
         bakedModelCollection = new EmptyBakedModelCollection();
-       this.cache = CacheBuilder.newBuilder().maximumSize(600).build(new CacheLoader<>() {
+        this.cache = CacheBuilder.newBuilder().maximumSize(600).build(new CacheLoader<>() {
             @Override
             public @NotNull FabricBakedModel load(@NotNull ItemStack stack) {
                 return bakedModelCollection.getToolPartModel(stack);
+            }
+        });
+        this.itemCache = CacheBuilder.newBuilder().maximumSize(600).build(new CacheLoader<>() {
+            @Override
+            public @NotNull FabricBakedModel load(@NotNull Item item) {
+                return bakedModelCollection.getToolPartModel(((ToolPartItem) item).getPart());
             }
         });
     }
 
     @Override
     public void emitItemQuads(ItemStack stack, Supplier<Random> randomSupplier, RenderContext context) {
-        cache.getUnchecked(stack).emitItemQuads(null, null, context);
+        if (stack.hasNbt()) {
+            cache.getUnchecked(stack).emitItemQuads(null, null, context);
+        } else {
+            itemCache.getUnchecked(stack.getItem()).emitItemQuads(null, null, context);
+        }
+
     }
 
     @Nullable
     @Override
     public BakedModel bake(ModelLoader loader, Function<SpriteIdentifier, Sprite> textureGetter, ModelBakeSettings rotationContainer, Identifier modelId) {
         this.bakedModelCollection = unbakedModelCollection.bakeModels(loader, textureGetter);
+        if (this.loader == null || this.loader != loader) {
+            this.loader = loader;
+            cache.invalidateAll();
+            itemCache.invalidateAll();
+        }
         return this;
     }
 }

--- a/src/main/java/com/sigmundgranaas/forgero/client/forgerotool/model/implementation/EmptyBakedModelCollection.java
+++ b/src/main/java/com/sigmundgranaas/forgero/client/forgerotool/model/implementation/EmptyBakedModelCollection.java
@@ -49,6 +49,7 @@ public class EmptyBakedModelCollection implements BakedModelCollection {
         return getEmptyModel(toolPart.getToolPartIdentifier());
     }
 
+
     private FabricBakedModel getEmptyModel(String identifier) {
         ForgeroInitializer.LOGGER.debug("Attempting to retrieve {} model from an Empty collection, return empty model", identifier);
         return new EmptyBakedModel();

--- a/src/main/java/com/sigmundgranaas/forgero/client/forgerotool/model/implementation/ModelCollectionImpl.java
+++ b/src/main/java/com/sigmundgranaas/forgero/client/forgerotool/model/implementation/ModelCollectionImpl.java
@@ -22,6 +22,7 @@ public class ModelCollectionImpl implements BakedModelCollection, UnbakedModelCo
     private final ToolModelAssembler toolAssembler;
     private final ToolPartModelAssembler toolPartModelAssembler;
     private Map<String, FabricBakedModel> bakedToolPartModels;
+    private ModelLoader loader;
 
 
     private ModelCollectionImpl(FabricToForgeroToolAdapter toolAdapter, FabricToForgeroToolPartAdapter toolPartAdapter) {
@@ -76,10 +77,12 @@ public class ModelCollectionImpl implements BakedModelCollection, UnbakedModelCo
         return toolPartModelAssembler.assembleToolPartModel(toolPart);
     }
 
+
     @Override
     public BakedModelCollection bakeModels(ModelLoader loader, Function<SpriteIdentifier, Sprite> textureGetter) {
-        if (bakedToolPartModels.isEmpty()) {
-            bakedToolPartModels = ToolPartModelFactory.createFactory(loader, textureGetter).createToolPartModels();
+        if (this.loader == null || this.loader != loader) {
+            this.loader = loader;
+            bakedToolPartModels = ToolPartModelFactory.createFactory(this.loader, textureGetter).createToolPartModels();
         }
         return this;
     }

--- a/src/main/java/com/sigmundgranaas/forgero/client/forgerotool/model/toolpart/BakedToolModel.java
+++ b/src/main/java/com/sigmundgranaas/forgero/client/forgerotool/model/toolpart/BakedToolModel.java
@@ -24,7 +24,7 @@ public class BakedToolModel implements FabricBakedModel {
 
     @Override
     public boolean isVanillaAdapter() {
-        return false;
+        return true;
     }
 
     @Override

--- a/src/main/java/com/sigmundgranaas/forgero/mixins/MiningToolItemMixin.java
+++ b/src/main/java/com/sigmundgranaas/forgero/mixins/MiningToolItemMixin.java
@@ -40,7 +40,9 @@ public class MiningToolItemMixin {
             ImmutableMultimap.Builder<EntityAttribute, EntityAttributeModifier> builder = ImmutableMultimap.builder();
 
             builder.put(EntityAttributes.GENERIC_ATTACK_DAMAGE, new EntityAttributeModifier(attributeModifiers.get(EntityAttributes.GENERIC_ATTACK_DAMAGE).stream().findFirst().get().getId(), "Tool modifier", this.attackDamage, EntityAttributeModifier.Operation.ADDITION));
-            builder.put(EntityAttributes.GENERIC_ATTACK_SPEED, new EntityAttributeModifier(attributeModifiers.get(EntityAttributes.GENERIC_ATTACK_SPEED).stream().findFirst().get().getId(), "Tool modifier", attackSpeed, EntityAttributeModifier.Operation.ADDITION));
+            if (attributeModifiers.get(EntityAttributes.GENERIC_ATTACK_SPEED).stream().findAny().isPresent()) {
+                builder.put(EntityAttributes.GENERIC_ATTACK_SPEED, new EntityAttributeModifier(attributeModifiers.get(EntityAttributes.GENERIC_ATTACK_SPEED).stream().findFirst().get().getId(), "Tool modifier", attackSpeed, EntityAttributeModifier.Operation.ADDITION));
+            }
             this.attributeModifiers = builder.build();
         }
     }

--- a/src/main/resources/data/forgero/gems/diamond.json
+++ b/src/main/resources/data/forgero/gems/diamond.json
@@ -26,7 +26,7 @@
         "type": "MINING_SPEED",
         "order": "END",
         "operation": "MULTIPLICATION",
-        "value": 10.0,
+        "value": 1.5,
         "condition": {
           "target": "BLOCK",
           "tag": [

--- a/src/main/resources/data/forgero/gems/ender.json
+++ b/src/main/resources/data/forgero/gems/ender.json
@@ -17,10 +17,10 @@
   "properties": {
     "attributes": [
       {
-        "type": "DURABILITY",
+        "type": "MINING_SPEED",
         "order": "BASE",
         "operation": "ADDITION",
-        "value": 1.0
+        "value": 2.0
       }
     ],
     "passive": [],

